apps/meteor/server/modules/notifications/notifications.module.ts

```diff
@@ -250,6 +251,17 @@ export class NotificationsModule {
 				return false;
 			}
 
+			if (
+				e === 'typing' ||
+				(e === 'user-activity' && Array.isArray(_activity) && (_activity.length === 0 || _activity.includes('user-typing')))
+			) {
+				api.broadcast('user.typing', {
```

Comment:
@RocketChat/chat-engine I've created this event to notify when the user is typing, but I'm not sure also if the best approach. Any suggestions on how to improve this? Thanks.

Reply:
The event itself is fine, but calling it from inside `allowWrite` is not ideal.

You should use `on` to listen to the event directly. This way you'll also only trigger it for the rooms that you are expecting it to.

```js
notifications.streamRoom.on(`${rid}/user-activity`, (username, activity) => {
  if (Array.isArray(activity) && (!activity.length || activity.includes('user-typing')) {
    api.broadcast('user.typing', {
      user: { username },
      isTyping: activity.includes('user-typing'),
      roomId: rid,
    });
  }
});
```